mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestAuthorizationFilter.java

```diff
@@ -157,6 +161,26 @@ public void filter(ContainerRequestContext requestContext) {
                 .toArray(String[]::new));
     }
 
+    private String resolveRequiredPermission(MCRRestRequiredPermission annotation,
+        ContainerRequestContext requestContext) {
+        final String requiredPermission = annotation.value();
+        final Class<? extends MCRRestRequiredPermissionResolver> requiredPermissionResolver = annotation.resolver();
+        if (requiredPermission.isEmpty()
+            && MCRRestRequiredPermissionDefaultResolver.class.equals(requiredPermissionResolver)) {
+            LogManager.getLogger().error("explicit value and resolver is not allowed");
+            throw new InternalServerErrorException();
+        } else if (!requiredPermission.isEmpty()) {
+            return requiredPermission;
+        } else {
+            try {
+                return requiredPermissionResolver.getDeclaredConstructor().newInstance()
+                    .resolvePermission(requestContext);
+            } catch (Exception e) {
+                throw new InternalServerErrorException(e);
+            }
+        }
```

Comment:
I think this could be simplified and have those cases reduced. See my other remarks